herbert_core/utilities/classes/@sigvar/sigvar.m

```diff
@@ -63,6 +63,35 @@
                     % class; in this case that a sigvar object be returned from
                     % a signvar object - a dummy operation)
                     obj = varargin{1};
+                elseif numel(varargin)==1 && isstruct(varargin{1})
+                    % local copy of input struct to fill in with missing
+                    % items as needed
+                    w = varargin{1};
+                    
+                    % if w does not have the signal field 's', it is an
+                    % error:
+                    wfields = fieldnames(w);
+                    if ~any(strcmp(wfields,'s'))
```

Comment:
`ismember(wfields, 's')` is probably a more efficient (and at the very least more clear) way of doing this.